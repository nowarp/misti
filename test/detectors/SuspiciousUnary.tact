fun bad1(): Int { return + 1; } // Bad
fun ok1(): Int { return - 1; } // OK
fun ok2(a: Int): Int { return a + 1; } // OK

contract C {
    field1: Int = 0;
    field2: Int = +1; // OK

    receive() {
        let a = +1; // Bad
        let b = -1; // OK
        a += 1; // OK
        a = b; // OK
        a = +b; // OK
        a = -b; // OK
        a = 1 + 1; // OK
        a = b + b; // OK

        self.field1 = +1; // Bad
        self.field1 = -1; // OK
        self.field1 += 1; // OK
        self.field1 = a + 1; // OK
    }
}
