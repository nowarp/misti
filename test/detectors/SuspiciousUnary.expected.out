[LOW] SuspiciousUnary: Suspicious unary expression
test/detectors/SuspiciousUnary.tact:1:26:
> 1 | fun bad1(): Int { return + 1; } // Bad
                               ^~~
  2 | fun ok1(): Int { return - 1; } // OK
+1 in that position may indicate typo
Help: Consider refactoring removing the unary `+`
See: https://nowarp.io/tools/misti/docs/detectors/SuspiciousUnary

[LOW] SuspiciousUnary: Suspicious unary expression
test/detectors/SuspiciousUnary.tact:10:17:
   9 |     receive() {
> 10 |         let a = +1; // Bad
                       ^~
  11 |         let b = -1; // OK
+1 in that position may indicate typo
Help: Consider refactoring removing the unary `+`
See: https://nowarp.io/tools/misti/docs/detectors/SuspiciousUnary

[LOW] SuspiciousUnary: Suspicious unary expression
test/detectors/SuspiciousUnary.tact:19:23:
  18 | 
> 19 |         self.field1 = +1; // Bad
                             ^~
  20 |         self.field1 = -1; // OK
+1 in that position may indicate typo
Help: Consider refactoring removing the unary `+`
See: https://nowarp.io/tools/misti/docs/detectors/SuspiciousUnary